tests/lib/v9-rule-migration/v9-rule-migration.js

```diff
@@ -0,0 +1,300 @@
+/**
+ * @fileoverview Tests for v9-rule-migration transform.
+ * @author Nitin Kumar
+ * MIT License
+ */
+
+"use strict";
+
+const path = require("path");
+const { applyTransform } = require("@hypermod/utils");
+const assert = require("assert");
+const sinon = require("sinon");
+const { normalizeLineEndings } = require("../../utils");
+const v9MigrationTransform = require("../../../lib/v9-rule-migration/v9-rule-migration");
+
+describe("v9 migration transform", () => {
+    it("should migrate deprecated context methods to new properties", async () => {
+        const result = await applyTransform(
+            v9MigrationTransform,
+            `
+            module.exports = {
+                create(context) {
+                    return {
+                        Program(node) {
+                            const sourceCode = context.getSourceCode();
+                            const cwd = context.getCwd();
+                            const filename = context.getFilename();
+                            const physicalFilename = context.getPhysicalFilename();
+                        },
+
+                        FunctionDeclaration(node) {
+                            const _sourceCode = context.getSourceCode();
+                            const _cwd = context.getCwd();
+                            const _filename = context.getFilename();
+                            const _physicalFilename = context.getPhysicalFilename();
+                        }
+                    };
+                }
+            };
+            `
+        );
+
+        assert.strictEqual(
+            normalizeLineEndings(result),
+            normalizeLineEndings(
+                `
+            module.exports = {
+                create(context) {
+                    const physicalFilename = context.physicalFilename ?? context.getPhysicalFilename();
+                    const filename = context.filename ?? context.getFilename();
+                    const cwd = context.cwd ?? context.getCwd();
+                    const sourceCode = context.sourceCode ?? context.getSourceCode();
+                    return {
+                        Program(node) {},
+
+                        FunctionDeclaration(node) {
+                            const _sourceCode = sourceCode;
+                            const _cwd = cwd;
+                            const _filename = filename;
+                            const _physicalFilename = physicalFilename;
+                        }
+                    };
+                }
+            };
+            `.trim()
+            )
+        );
+    });
+
+    it("should migrate deprecated context methods to SourceCode", async () => {
+        const result = await applyTransform(
+            v9MigrationTransform,
+            `
+            module.exports = {
+                create(context) {
+                    return {
+                        Program(node) {
+                            const sourceCodeText = context.getSource();
```

Comment:
```suggestion
                            const sourceCode = context.getSource();
```

Is it intentionally written here as `sourceCodeText`?

Reply:
Yes. This method returns a string.

Reply:
ops, I misread it to `getSourceCode()`